nuleaf-source/database/posts.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Posts database access object. Opinionated wrapper around mongoose.
+ *
+ * Handles requests to posts collection.
+ */
+
+/* jslint node: true */
+'use strict';
+
+var Helper = require('../util/helper');
+
+var Post = require('../models/post');
+
+
+/**
+ * Finds posts matching conditions and returns a collection of posts.
+ * @param {Object} Conditions:
+ *           name          : Filter for posts with this name.
+ *           content       : Filter for posts with this content.
+ *           author        : Filter for posts with this author.
+ *           date_created  : Filter for posts created on this date.
+ *           date_published: Filter for posts published on this date.
+ *           date_modified : Filter for posts modified on this date.
+ *           skip          : Return a certain number of results after a certain number documents.
+ *           limit         : Used to specify the maximum number of results to be returned.
+ *
+ * @return {Error}, {Array} Array of posts, or empty if none found matching conditions.
+ */
+exports.find = function(conditions, callback) {
+  if (typeof conditions === 'function') {
+ 	  callback = conditions;
+ 	  conditions = {};
+  }
+
+  var skip = +conditions.skip;
+  var limit = +conditions.limit;
+
+  delete conditions.skip;
+  delete conditions.limit;
+
+  var _condiitons = buildConditions(conditions);
+
+  if ( Post.postSchema.paths.hasOwnProperty(conditions.sortBy)) {
+    var sort = {};
+    sort[conditions.sortBy] = +conditions.sort;
+    delete conditions.sortBy;
+    delete conditions.sort;
+    Post.find(_conditions).skip(skip).limit(limit).sort(sort).exec(callback);
```

Comment:
`Post.find(_conditions).skip(skip).limit(limit)` returns a `QueryBuilder` object which you can call exec on later.

We should try to be as [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself) as possible.
```
query = Post.find(_conditions).skip(skip).limit(limit);
if (...) query = query.sort(sort);

query.exec(callback);
```